src/evict/evict_page.c

```diff
@@ -8,10 +8,53 @@
 
 #include "wt_internal.h"
 
+static void __evict_enter(WT_SESSION_IMPL *);
+static void __evict_leave(WT_SESSION_IMPL *);
 static int __evict_page_clean_update(WT_SESSION_IMPL *, WT_REF *, uint32_t);
 static int __evict_page_dirty_update(WT_SESSION_IMPL *, WT_REF *, uint32_t);
 static int __evict_review(WT_SESSION_IMPL *, WT_REF *, uint32_t, bool *);
 
+/*
```

Comment:
If I remember correctly we had made reconciliation non-reentrant for similar reasons, maybe couple of years back. We had issues with reconciliation repeatedly calling itself when using lookaside file. Are we sort of duplicating ```WT_SESSION_NO_RECONCILE``` with ```WT_SESSION_EVICTING```. Keith B. did that work, I am not sure if he is available for a review. @keitharnoldsmith can you have a look at ```WT_SESSION_NO_RECONCILE``` and see if it serves a different purpose than what you want to solve here.

Reply:
Thanks @sulabhM.  You raise a good point.  It seems like there is enough synergy here that we may be able to combine them in some way. I know that the current use of `WT_SESSION_NO_RECONCILE` isn't sufficient to prevent us from re-entering eviction, but it is worth spending a bit of time looking.  I'll see what I can come up with tomorrow.